pgrx-macros/src/rewriter.rs

```diff
@@ -49,10 +49,12 @@ pub fn item_fn_without_rewrite(mut func: ItemFn) -> syn::Result<proc_macro2::Tok
         panic!("#[pg_guard] for function with generic parameters must not be combined with #[no_mangle]");
     }
 
+    attrs.push(syn::parse_quote!(#[automatically_derived]));
```

Comment:
@workingjubilee https://stackoverflow.com/questions/51481551/what-does-automatically-derived-mean -- see also serde [implementation](https://github.com/search?q=repo%3Aserde-rs%2Fserde+automatically_derived&type=code)

Reply:
I know what it means. I have looked into the compiler. My question is why are you suggesting I add this, specifically. What does it actually change? I don't care if serde does it. Just because the build tool's name is cargo does not mean we should actually cargo cult things.

Reply:
I agree about curgo culting.  My issue is that I would like MY code to be clean, and I rely on Clippy to catch issues for me. Clippy has done an awesome job **for me** in this regard.  The problem is that any generated code is not as easy to annotate with attributes. So instead, I am forced to place just the pgrx generated code into separate modules/files and add module-level attributes like `#![allow(clippy::used_underscore_binding)]` so that Clippy does not complain about it -- making code management harder. You may say this is a limitation of Clippy (although I could see why they choose to do it), but at the end of the day, if no other auto-generated code cause Clippy warnings, and PGRX does, it affects your users, and creates un-needed frictions. It is far easier to simply include extra invisible attributes to the generated code than to cause each one of your user to deal with it. Just like all other macro-generating crates do.

Reply:
Report these bugs to clippy **first** before coming to me with them.

Reply:
And stop asking me to add these annotations without first trying to actually fix the issue that gets linted on.

Reply:
If you really can be bothered with neither of those, then there is no use for filing an allowance for every single lint individually. Simply add `allow(clippy::all)`.

Reply:
You're talking about friction because of a few lints, while I'm spending my work days trying to fix high-priority soundness issues that also reduce the performance of the real generated artifacts. In other words, they undermine the very point of the pgrx library. Clippy doesn't exactly detect that. I don't really care if the generated code is pretty or not or generates a lot of warnings while these deep, structural issues remain. I think you *should* be alarmed at how bad the code it generates is. I think people *should* experience friction while using it, until the actual problems get actually solved, instead of merely papered over. I don't like painting a pretty picture and hanging it up over a *rotting inside*. It feels like lying.

And meanwhile clippy has over 400 false positive issues open. https://github.com/rust-lang/rust-clippy/issues?q=is%3Aissue+is%3Aopen+label%3AI-false-positive

Reply:
This attribute, if added, would probably increase the number of warnings generated by rustc.

https://github.com/rust-lang/rust/blob/4bc39f028d14c24b04dd17dc425432c6ec354536/tests/ui/feature-gates/issue-43106-gating-of-builtin-attrs.rs#L266-L273

Reply:
@nyurik This is why I asked that question. I want to understand what actually happens. I want to know how rustc will use this. And apparently the answer is mostly in ways that are liable to impair diagnostics:

https://github.com/rust-lang/rust/blob/4bc39f028d14c24b04dd17dc425432c6ec354536/compiler/rustc_trait_selection/src/traits/error_reporting/type_err_ctxt_ext.rs#L2104-L2114

And that is only when it doesn't generate warnings outright:

https://github.com/rust-lang/rust/blob/4bc39f028d14c24b04dd17dc425432c6ec354536/tests/ui/feature-gates/issue-43106-gating-of-builtin-attrs.rs#L266-L279

So no, I don't really want to add `#[automatically_derived]` to the code that pgrx emits on top of a function?